Core/Generators/Rust/RustGenerator.cs

```diff
@@ -116,6 +119,58 @@ private void WriteConstDefinition(IndentedStringBuilder builder, ConstDefinition
                 $"pub const {MakeConstIdent(d.Name)}: {TypeName(d.Value.Type, OwnershipType.Constant)} = {EmitLiteral(d.Value)};");
         }
 
+        private void WriteFlagsEnumDefinition(IndentedStringBuilder builder, EnumDefinition d)
+        {
+            // main definition
+            var name = MakeDefIdent(d.Name);
+            builder.CodeBlock("::bitflags::bitflags!", _tab, () =>
+            {
+                builder.CodeBlock($"pub struct {name}: u32", _tab, () =>
+                {
+                    foreach (var m in d.Members)
+                    {
+                        WriteDocumentation(builder, m.Documentation);
+                        WriteDeprecation(builder, m.DeprecatedAttribute);
+                        builder.AppendLine($"const {MakeConstIdent(m.Name)} = {m.ConstantValue};");
+                    }
+                });
+            }).AppendLine();
+
+            // sub record
+            builder.CodeBlock($"impl<'raw> ::bebop::SubRecord<'raw> for {name}", _tab, () =>
+            {
+                builder
+                    .AppendLine("const MIN_SERIALIZED_SIZE: usize = ::bebop::ENUM_SIZE;")
```

Comment:
Can we perhaps make some of this a function since it is very similar to the other enum definition.